test/playback-watcher.test.js

```diff
@@ -42,6 +42,97 @@ QUnit.module('PlaybackWatcher', {
   }
 });
 
+QUnit.test('skips over gap at beginning of stream if player is paused', function(assert) {
+  let vhsGapSkipEvents = 0;
+  let hlsGapSkipEvents = 0;
+
+  this.player.autoplay(true);
+
+  this.player.tech_.on('usage', (event) => {
+    if (event.name === 'vhs-gap-skip') {
+      vhsGapSkipEvents++;
+    }
+    if (event.name === 'hls-gap-skip') {
+      hlsGapSkipEvents++;
+    }
+  });
+
+  // set an arbitrary source
+  this.player.src({
+    src: 'master.m3u8',
+    type: 'application/vnd.apple.mpegurl'
+  });
+
+  // start playback normally
+  this.player.tech_.triggerReady();
+  this.clock.tick(1);
+  standardXHRResponse(this.requests.shift());
+  openMediaSource(this.player, this.clock);
+  this.player.tech_.trigger('play');
+  this.player.tech_.trigger('waiting');
+  // create a buffer with a gap of 2 seconds at beginning of stream
+  this.player.tech_.buffered = () => videojs.createTimeRanges([[2, 10]]);
+  // Playback watcher loop runs on a 250ms clock and needs run 6 consecutive stall checks before skipping the gap
+  this.clock.tick(250 * 6);
+  // Need to wait for the duration of the gap
+  this.clock.tick(2000);
+
+  assert.equal(vhsGapSkipEvents, 1, 'there is one skipped gap');
+  assert.equal(hlsGapSkipEvents, 1, 'there is one skipped gap');
+
+  // check that player jumped the gap
+  assert.equal(
+    Math.round(this.player.currentTime()),
+    2, 'Player seeked over gap after timer'
+  );
+});
+
+QUnit.test('Does NOT skip over gap at beginning of stream if player is playing', function(assert) {
```

Comment:
Technically, I think we'd want this case to be handled by the gap skipper too, but the gap skipper is not set up yet at the time of these events. I'm not sure we'd run into it, since the player is marked as playing (paused set to false) before the source is set and ready is triggered.

I think if we ever run into a case where the player is playing before gap skipper is setup, we may want to remove the check for `if (this.tech_.paused()) {`, but until we run into that case, I think it's safer to leave that check there.

It may be best to remove this test and instead just add a test that checks that we don't listen to future `play` events after the first one (since we use the `one` listener) by checking that we don't reset the monitor if we get `play` events in-between.

Reply:
Removed the test and added a new test which proves that additional `play` events do not speed up the timing of the  playback monitor. Also confirmed that using `.on('play')` results in this test failing.